src/Maker/MakeEntity.php

```diff
@@ -35,49 +60,706 @@ public static function getCommandName(): string
+                    EntityRelation::MANY_TO_ONE,
+                    $targetEntityClass,
+                    $generatedEntityClass
+                );
+                $relation->setInverseProperty($newFieldName);
+
+                $io->comment(sprintf(
+                    'A new property will also be added to the <comment>%s</comment> class so that you can access and set the related <comment>%s</comment> object from it.',
+                    Str::getShortClassName($relation->getOwningClass()),
+                    Str::getShortClassName($relation->getInverseClass())
+                ));
+                $relation->setOwningProperty($askFieldName(
+                    $relation->getOwningClass(),
+                    Str::asLowerCamelCase(Str::getShortClassName($relation->getInverseClass()))
+                ));
+
+                $relation->setIsNullable($askIsNullable(
+                    $relation->getOwningProperty(),
+                    $relation->getOwningClass()
+                ));
+
+                if (!$relation->isNullable()) {
+                    $relation->setOrphanRemoval($askOrphanRemoval(
+                        $relation->getOwningClass(),
+                        $relation->getInverseClass()
+                    ));
+                }
+
+                break;
+            case EntityRelation::MANY_TO_MANY:
+                $relation = new EntityRelation(
+                    EntityRelation::MANY_TO_MANY,
+                    $generatedEntityClass,
+                    $targetEntityClass
+                );
+                $relation->setOwningProperty($newFieldName);
+
+                $setMapInverseSide($relation);
+                if ($relation->getMapInverseRelation()) {
+                    $io->comment(sprintf(
+                        'A new property will also be added to the <comment>%s</comment> class so that you can access the related <comment>%s</comment> objects from it.',
+                        Str::getShortClassName($relation->getInverseClass()),
+                        Str::getShortClassName($relation->getOwningClass())
+                    ));
+                    $relation->setInverseProperty($askFieldName(
+                        $relation->getInverseClass(),
+                        Str::singularCamelCaseToPluralCamelCase(Str::getShortClassName($relation->getOwningClass()))
+                    ));
+                }
+
+                break;
+            case EntityRelation::ONE_TO_ONE:
+                $relation = new EntityRelation(
+                    EntityRelation::ONE_TO_ONE,
+                    $generatedEntityClass,
+                    $targetEntityClass
+                );
+                $relation->setOwningProperty($newFieldName);
+
+                $relation->setIsNullable($askIsNullable(
+                    $relation->getOwningProperty(),
+                    $relation->getOwningClass()
+                ));
+
+                $setMapInverseSide($relation);
```

Comment:
you should never generate a bidirectional one-to-one relation IMO. they are a performance killer (the inverse side cannot use a lazy-loading proxy, as it does not have the id in its own table, so it *always* creates an extra query each time)

Reply:
I love having feedback at this deep of a level - awesome :).

I just tested, and it's not *quite* as you say: only *1* query is made, but it's a join query. But I think it's basically the same issue: if you query directly for the inverse entity, you query for 2x the data, even if you don't need it. So yea, I'm going to reverse this: no inverse relation for OneToOne